runtime/http_server_apis.md

```diff
@@ -206,6 +215,79 @@ serveTls(handler, {
+
+Currently Deno supports gzip and brotli compression. A body is automatically
+compressed if the following conditions are true:
+
+- The request has an
+  [`Accept-Encoding`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding)
+  header which indicates the requestor supports `br` for brotli or `gzip`. Deno
+  will respect the preference of the
+  [quality value](https://developer.mozilla.org/en-US/docs/Glossary/Quality_values)
+  in the header.
+- The response includes a
+  [`Content-Type`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)
+  which is considered compressible. (The list is derived from
+  [`jshttp/mime-db`](https://github.com/jshttp/mime-db/blob/master/db.json) with
+  the actual list in the
+  [code](https://github.com/denoland/deno/blob/$CLI_VERSION/ext/http/compressible.rs).)
+- The response body is greater than 20 bytes.
+
+When the response body is compressed, Deno will set the
+[`Content-Encoding`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding)
+header to reflect the encoding as well as ensure the
+[`Vary`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary) header
+is adjusted or added to indicate what request headers affected the response.
+
+#### When is compression skipped?
+
+In addition to the logic above, there are a few other reasons why a response
+won't be compressed automatically:
+
+- The response body is a stream. Currently only _static_ response bodies are
+  supported. We will add streaming support in the future.
+- The response contains a `Content-Encoding` header. This indicates your server
+  has done some form of encoding.
+- The response contains a
+  [`Content-Range`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range)
+  header. This indicates that your server is responding to a range request,
+  where the bytes and ranges are negotiated outside of the control of the
+  internals to Deno.
+- The response has a
+  [`Cache-Control`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)
+  header which contains a
+  [`no-transform`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#other)
+  value. This indicates that your server doesn't want Deno or any downstream
+  proxies to modify the response.
+
+#### What happens to an `ETag` header?
+
+When you set an
+[`ETag`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) what is
+not a weak validator and the body is compressed, Deno will change this to a weak
+validator (`W/`). This is to ensure the proper behavior of clients and
+downstream proxy services when validating the "freshness" of the content of the
+response body.
+
+#### What happens to a `Content-Length` header?
+
+If the response contains a
+[`Content-Length`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Length)
+header, it will be unset if the body is compressed, as the content length will
+have changed.
```

Comment:
This can be removed. The result will still have a `Content-Length` (hyper will just set it to the length of the compressed value, right?)

Reply:
Ah yeah, right.